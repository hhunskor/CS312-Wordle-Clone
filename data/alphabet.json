[
  {
    "letter": "A",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "B",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "C",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "D",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "E",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "F",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "G",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "H",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "I",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "J",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "K",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "L",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "M",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "N",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "O",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "P",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "Q",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "R",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "S",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "T",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "U",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "V",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "W",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "X",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "Y",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  },
  {
    "letter": "Z",
    "guessed": false,
    "guessedInWord": false,
    "guessedInPlace": false
  }
]
