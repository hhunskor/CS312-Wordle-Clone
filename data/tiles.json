[
  {
    "tile": 0,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 1,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 2,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 3,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 4,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 5,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 6,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 7,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 8,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 9,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 10,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 11,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 12,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 13,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 14,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 15,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 16,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 17,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 18,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 19,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 20,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 21,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 22,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 23,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 24,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 25,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 26,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 27,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 28,
    "letter": "",
    "color": "black"
  },
  {
    "tile": 29,
    "letter": "",
    "color": "black"
  }
]
