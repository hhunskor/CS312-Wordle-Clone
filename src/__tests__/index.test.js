import { render, screen, fireEvent } from "@testing-library/react";
import Main from "../pages/index";
import words from "../../data/words.json";
import tiles from "../../data/tiles.json";
import alphabet from "../../data/alphabet.json";
/**
 * This is a placeholder test that is here only to make sure `npm test` will not choke because there are no tests. It should be removed as soon as you write a real test.
 */

describe("Homepage visibility", () => {
  let testAlphabet;
  let testTiles;

  beforeEach(() => {
    testAlphabet = alphabet.map((x) => ({ ...x }));
    testTiles = tiles.map((x) => ({ ...x }));
  });

  test("Smoke test", () => {
    render(
      <Main
        arrayWords={words}
        correctWord={"guard"}
        alphabet={testAlphabet}
        setAlphabet={jest.fn()}
        tiles={testTiles}
        setTiles={jest.fn()}
      />
    );
  });
  test("Title Visible", () => {
    render(
      <Main
        arrayWords={words}
        correctWord={"guard"}
        alphabet={testAlphabet}
        setAlphabet={jest.fn()}
        tiles={testTiles}
        setTiles={jest.fn()}
      />
    );
    expect(screen.getByText("A Middlebury Wordle Project")).toBeVisible();
  });
  test("Snapshot test", () => {
    const { container } = render(
      <Main
        arrayWords={words}
        correctWord={"guard"}
        alphabet={testAlphabet}
        setAlphabet={jest.fn()}
        tiles={testTiles}
        setTiles={jest.fn()}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Submit button updates", () => {
  let testAlphabet;
  let testTiles;

  const firstGuess = "guess";
  const updatedTiles = tiles.map((x) => {
    if (x.tile === 0) return { tile: 0, letter: "G", color: "green" };
    else if (x.tile === 1) return { tile: 1, letter: "U", color: "green" };
    else if (x.tile === 2) return { tile: 2, letter: "E", color: "gray" };
    else if (x.tile === 3) return { tile: 3, letter: "S", color: "gray" };
    else if (x.tile === 4) return { tile: 4, letter: "S", color: "gray" };
    else if (x.tile > 4) return x;
  });

  const updatedAlphabet = alphabet.map((x) => {
    if (x.letter === "G")
      {return {
        letter: "G",
        guessed: true,
        guessedInWord: true,
        guessedInPlace: true,
      };}
    else if (x.letter === "U")
      {return {
        letter: "U",
        guessed: true,
        guessedInWord: true,
        guessedInPlace: true,
      };}
    else if (x.letter === "E")
      {return {
        letter: "E",
        guessed: true,
        guessedInWord: false,
        guessedInPlace: false,
      };}
    else if (x.letter === "S")
      {return {
        letter: "S",
        guessed: true,
        guessedInWord: false,
        guessedInPlace: false,
      };}
    else return x;
  });

  beforeEach(() => {
    testAlphabet = alphabet.map((x) => ({ ...x }));
    testTiles = tiles.map((x) => ({ ...x }));
  });

  afterEach(() => {
    jest.clearAllMocks;
  });

  const setTilesMock = jest.fn();
  const setAlphabetMock = jest.fn();

  test("Submit button updates tiles.json", () => {
    render(
      <Main
        arrayWords={words}
        correctWord={"GUARD"}
        alphabet={testAlphabet}
        setAlphabet={setAlphabetMock}
        tiles={testTiles}
        setTiles={setTilesMock}
      />
    );

    const inputBox = screen.getByTestId("Input");
    const submit = screen.getByTestId("Submit");

    fireEvent.change(inputBox, {
      target: { value: firstGuess },
    });

    expect(inputBox.value).toBe("guess");

    fireEvent.click(submit);
    const newTiles = setTilesMock.mock.calls[0][0];
    expect(newTiles).toEqual(updatedTiles);
  });

  test("Submit button updates alphabet.json", () => {
    render(
      <Main
        arrayWords={words}
        correctWord={"GUARD"}
        alphabet={testAlphabet}
        setAlphabet={setAlphabetMock}
        tiles={testTiles}
        setTiles={setTilesMock}
      />
    );

    const inputBox = screen.getByTestId("Input");
    const submit = screen.getByTestId("Submit");

    fireEvent.change(inputBox, {
      target: { value: firstGuess },
    });

    expect(inputBox.value).toBe("guess");

    fireEvent.click(submit);
    const newAlphabet = setAlphabetMock.mock.calls[0][0];
    expect(newAlphabet).toEqual(updatedAlphabet);
  });
});
