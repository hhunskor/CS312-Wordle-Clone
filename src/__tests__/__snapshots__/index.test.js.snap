// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage visibility Snapshot test 1`] = `
<div
  class="container"
>
  <main>
    <h1
      class="title"
    >
      A Middlebury Wordle Project
    </h1>
    <p>
      Enter a 5 letter word in the input box to try to guess the correct word.
    </p>
    <div>
      <div
        id="grid"
      >
        <div
          class="row"
        >
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
          <div
            class="initBox"
          >
             
             
          </div>
        </div>
      </div>
    </div>
    <input
      data-testid="Input"
      maxlength="5"
      placeholder="Guess Here"
      type="text"
    />
    <button
      data-testid="Submit"
      disabled=""
      type="Submit"
    >
      Submit
    </button>
    <p>
      Start a new game by refreshing the page.
    </p>
  </main>
  <div
    class="keyboard"
  >
    <div
      class="keyRow"
    >
      <button
        class="grayKey"
      >
        Q
      </button>
      <button
        class="grayKey"
      >
        W
      </button>
      <button
        class="grayKey"
      >
        E
      </button>
      <button
        class="grayKey"
      >
        R
      </button>
      <button
        class="grayKey"
      >
        T
      </button>
      <button
        class="grayKey"
      >
        Y
      </button>
      <button
        class="grayKey"
      >
        U
      </button>
      <button
        class="grayKey"
      >
        I
      </button>
      <button
        class="grayKey"
      >
        O
      </button>
      <button
        class="grayKey"
      >
        P
      </button>
    </div>
    <div
      class="keyRow"
    >
      <button
        class="grayKey"
      >
        A
      </button>
      <button
        class="grayKey"
      >
        S
      </button>
      <button
        class="grayKey"
      >
        D
      </button>
      <button
        class="grayKey"
      >
        F
      </button>
      <button
        class="grayKey"
      >
        G
      </button>
      <button
        class="grayKey"
      >
        H
      </button>
      <button
        class="grayKey"
      >
        J
      </button>
      <button
        class="grayKey"
      >
        K
      </button>
      <button
        class="grayKey"
      >
        L
      </button>
    </div>
    <div
      class="keyRow"
    >
      <button
        class="grayKey"
      >
        Z
      </button>
      <button
        class="grayKey"
      >
        X
      </button>
      <button
        class="grayKey"
      >
        C
      </button>
      <button
        class="grayKey"
      >
        V
      </button>
      <button
        class="grayKey"
      >
        B
      </button>
      <button
        class="grayKey"
      >
        N
      </button>
      <button
        class="grayKey"
      >
        M
      </button>
    </div>
  </div>
  <footer>
    A 312 project | Authors: Team Frogfish Hannah, Addison, Wright, Hayden, Lucas, Jared, and Lizzie
     
  </footer>
</div>
`;
